iconservice/base/type_converter.py

```diff
@@ -278,15 +279,30 @@ def adjust_params_to_method(func: callable, kw_params: dict, remove_invalid_para
             if remove_invalid_param:
                 for key in invalid_keys:
                     del kw_params[key]
+                if len(kw_params) == 0:
+                    # input invalid params only
+                    raise InvalidParamsException(f"There is no valid parameters")
             else:
-                raise InvalidParamsException(f"Invalid parameter name '{invalid_keys}'")
+                raise InvalidParamsException(f"Invalid parameters '{invalid_keys}'")
 
         # check required argument is exist in user input
         for param_name, param_type in hints.items():
             if param_name == "self" or param_name == "cls":
                 continue
 
-            param = kw_params.get(param_name, None)
+            try:
+                param = kw_params[param_name]
+            except KeyError:
+                # has no input for this parameter
+                try:
+                    default_args[param_name]
+                except KeyError:
+                    # has no default value
+                    raise InvalidParamsException(f"There is no '{param_name}' parameter")
+                # has default value. pass type converting
```

Comment:
How about this code? It can reduce the use of the exception.
```
            try:
                param = kw_params.get(param_name, default_args[param_name])
            except KeyError:
                raise InvalidParamsException(f"There is no '{param_name}' parameter")

```

Reply:
Nice approach!!

Your code raise exception correctly but try to convert default value.

Reply:
Using try twice could be a more efficient way As not converting the default value.